File: 2022/2/java/A.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class A {

    private static final int OPP_ENC = 'A';
    private static final int ME_ENC = 'X';

    private static final int DRAW_SCORE = 3;
    private static final int WIN_SCORE = 6;
    private static final int LOSE_SCORE = 0;

    private static int compute(int opp, int me) {
        // Draw
        if (opp == me) return DRAW_SCORE + me;

        // Win cases
        if (opp == 1 && me == 2 ||
            opp == 2 && me == 3 ||
            opp == 3 && me == 1)
          return WIN_SCORE + me;

        // All other cases lose
        return LOSE_SCORE + me;
    }

    // O(N) time, O(1) space
    private static void solve(String[] rawInput) {
        // Guard
        if (rawInput == null || rawInput.length < 1) {
            throw new Error("Raw input is NULL!");
        }

        // System.out.println("testing input...");
        // for (String inp : rawInput) {
        //     System.out.println("Line: " + inp);
        // }

        int total = 0;
        for (String inp : rawInput) {
            // Further split the input by space.
            String[] vals = inp.split(" ");
            int opp = vals[0].charAt(0) - OPP_ENC + 1;
            int me = vals[1].charAt(0) - ME_ENC + 1;
            total += compute(opp, me);
        }

        System.out.println("Total: " + total);
    }

    private static void run(String filename) {
        // Guard
        if (filename == null || filename.isEmpty()) {
            throw new Error("Filename is NULL!");
        }

        // Open file and read it.
        String content = null;
        try {
            content = Files.readString(Paths.get(filename));
        } catch (IOException ex) {
            ex.printStackTrace();
        }

        // Now split this in multiple rows, if exist.
        String[] rawInput = content.split(System.lineSeparator());

        solve(rawInput);
    }

    private static final String DEFAULT_INPUT_FILENAME = "input";

    public static void main(String[] args) {
        // Parse argument.
        String filename = null;
        filename = args != null && args.length > 0 ? args[0] : DEFAULT_INPUT_FILENAME;

        // Run the code.
        run(filename);
    }
}

File: 2022/2/py/a.py
import argparse

# Rock, paper, scissors
OPP_ENC: int = ord('A')
ME_ENC: int = ord('X')

DRAW_SCORE: int = 3
WIN_SCORE: int = 6
LOSE_SCORE: int = 0

def compute(opp: int, me: int) -> int:
  # Draw case
  if opp == me:
    return DRAW_SCORE + me

  # Win cases
  if me == 2 and opp == 1 or \
     me == 3 and opp == 2 or \
     me == 1 and opp == 3:
    return WIN_SCORE + me

  # All other cases are losers
  return LOSE_SCORE + me


def solve(raw_input: list[str]):
  inp_pairs: list[str] = raw_input.split('\n')
  total: int = 0
  for inp_pair in inp_pairs:
    vals: list[str] = inp_pair.split()
    opp: int = ord(vals[0]) - OPP_ENC + 1
    me: int = ord(vals[1]) - ME_ENC + 1
    total += compute(opp, me)

  print(f"{total=}")


def main(filename: str):
  # Read file and convert to ints
  input = []
  with open(filename) as file:
    input = file.read()
    solve(input)

if __name__ == '__main__':

  # Simple argparse for the input
  parser = argparse.ArgumentParser()
  parser.add_argument("filename", type=str, nargs="?", default="input",
        help="Enter the name of the file (default = input)")
  parser.add_argument("-i", "--interactive", default="false", action="store_true", dest="interactive",
        help="If flag is set, the input is interactive")
  args = parser.parse_args()

  # Run the program
  main(args.filename)